39_component_life_cycle/parettool/src/App.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { useEffect, useState } from 'react';
+import Header from './Header/Header';
+import Products from './products/Products';
+import './App.css';
+
+const App = () => {
+  let productsAll = [];
```

Comment:
I don't see any reason to have this variable, it is enough:

`const [products, setProducts] = useState([]);`

Reply:
Thank you. I think I understood.   
  const [products, setProducts] = useState([]);

  const apiGet = () => {
    fetch('https://dummyjson.com/products')
    .then((res) => res.json())
    .then((data) => setProducts(data.products))
  };